src/components/Dashboard/Dashboard.js

```diff
@@ -42,101 +51,96 @@ const dashboardMap = {
+        return dashboardMap[mode]
     }
 
-    componentDidMount() {
-        window.onresize = debounce(
-            () => this.props.setWindowHeight(window.innerHeight),
-            300
+    if (dashboardsIsEmpty) {
+        return (
+            <>
+                <DashboardsBar />
+                <DashboardVerticalOffset />
+                <NoContentMessage
+                    text={i18n.t(
+                        'No dashboards found. Use the + button to create a new dashboard.'
+                    )}
+                />
+            </>
         )
-        this.setDashboard()
     }
 
-    componentDidUpdate() {
-        this.setDashboard()
+    if (id === NON_EXISTING_DASHBOARD_ID) {
+        return (
+            <>
+                <DashboardsBar />
+                <DashboardVerticalOffset />
+                <NoContentMessage
+                    text={i18n.t('Requested dashboard not found')}
+                />
+            </>
+        )
     }
 
-    render() {
-        const { id, mode, dashboardsLoaded, dashboardsIsEmpty } = this.props
-
-        if (!dashboardsLoaded || id === null) {
-            return (
-                <Layer translucent>
-                    <CenteredContent>
-                        <CircularLoader />
-                    </CenteredContent>
-                </Layer>
-            )
-        }
-
-        if (mode === NEW) {
-            return dashboardMap[mode]
-        }
-
-        if (dashboardsIsEmpty) {
-            return (
-                <>
-                    <DashboardsBar />
-                    <DashboardVerticalOffset />
-                    <NoContentMessage
-                        text={i18n.t(
-                            'No dashboards found. Use the + button to create a new dashboard.'
-                        )}
-                    />
-                </>
-            )
-        }
-
-        if (id === NON_EXISTING_DASHBOARD_ID) {
-            return (
-                <>
-                    <DashboardsBar />
-                    <DashboardVerticalOffset />
-                    <NoContentMessage
-                        text={i18n.t('Requested dashboard not found')}
-                    />
-                </>
-            )
-        }
-
-        return dashboardMap[mode]
-    }
+    return dashboardMap[mode]
 }
 
 Dashboard.propTypes = {
     dashboardsIsEmpty: PropTypes.bool,
     dashboardsLoaded: PropTypes.bool,
     id: PropTypes.string,
-    match: PropTypes.object,
     mode: PropTypes.string,
+    routeId: PropTypes.string,
     selectDashboard: PropTypes.func,
     setWindowHeight: PropTypes.func,
 }
 
-const mapStateToProps = state => {
+const mapStateToProps = (state, ownProps) => {
     const dashboards = sGetAllDashboards(state)
     return {
         dashboardsIsEmpty: isEmpty(dashboards),
         dashboardsLoaded: !sDashboardsIsFetching(state),
         id: sGetSelectedId(state),
+        routeId: ownProps.match.params.dashboardId || null,
```

Comment:
This will break with a reference error if `match` or `params` is undefined - we should either define an explicit shape requirement in `propTypes` or use optional chaining (`ownProps.match?.params?.dashboardId`) here

Reply:
`match` could also be just a regular prop, and use `const routeId = match?.params?.dashboardId || null` could be at the top of the component body

Reply:
I will add that. I did move `match.params.dashboardId` from componentDidMount. It had been like that since the original dashboard code, so I figured it will always be defined. But best to be sure.